### PR TITLE
Add batch processing function

### PR DESCRIPTION
**Pull request by berland**

### CODE REVIEW

**Comment by berland** [comment]
This PR is there to help #77 

Needs more thinking on the data structure sent as the `batch` argument.

**Comment by berland** [comment]
This might be ready unless someone comes with a better data structure for the list of batch commands, that simplifies syntax.

File: src/fmu/ensemble/ensemble.py [hunk 1]
```diff
@@ -839,6 +855,26 @@ def drop(self, localpath, **kwargs):
...
+
+        Args:
+            batch (list): Each list element is a dictionary with one key,
+                being a function names, value pr key is a dict with keyword
+                arguments to be supplied to each function.
+        Returns:
+            ScratchEnsemble: This ensemble object (self), for it
+                to be picked up by ProcessPoolExecutor and pickling.
+        """
+        for realidx, realization in self._realizations.items():
```
Comment (wouterjdb) [inline src/fmu/ensemble/ensemble.py]: use `.values()`

File: src/fmu/ensemble/ensembleset.py [hunk 2]
```diff
@@ -431,6 +447,23 @@ def drop(self, localpath, **kwargs):
...
+        but this enables more efficient concurrency. It is meant
+        to be used for functions that modifies the realization
+        object, not for functions that returns a dataframe already.
+
+        Args:
+            batch (list): Each list element is a dictionary with one key,
+                being a function names, value pr key is a dict with keyword
+                arguments to be supplied to each function.
+        """
+        for ens_name, ensemble in self._ensembles.items():
```
Comment (wouterjdb) [inline src/fmu/ensemble/ensembleset.py]: Use `.values()`

File: src/fmu/ensemble/ensemble.py (hunk 1 above)
Comment (berland) [reply src/fmu/ensemble/ensemble.py]: That would not give me the realization index.

File: src/fmu/ensemble/ensembleset.py (hunk 2 above)
Comment (berland) [reply src/fmu/ensemble/ensembleset.py]: That would not return the ens_name.

File: src/fmu/ensemble/ensemble.py (hunk 1 above)
Comment (berland) [reply src/fmu/ensemble/ensemble.py]: But you are right, I am not using the index...

**Comment by berland** [comment]
Requested changes have been commited. Ready to merge.

File: src/fmu/ensemble/ensemble.py [hunk 3]
```diff
@@ -350,16 +366,76 @@ def to_virtual(self, name=None):
...
+    def manifest(self, manifest):
+        """Set the manifest of the ensemble. The manifest
+        is nothing but a Python dictionary with unspecified
+        content
+
+        Args:
+            manifest: dict or str. If dict, it is used as is, if str it
+                is assumed to be a filename with YAML syntax which is
+                parsed into a dict and stored as dict
+        """
```
Comment (wouterjdb) [inline src/fmu/ensemble/ensemble.py]: Returns: Nothing

Comment (berland) [reply src/fmu/ensemble/ensemble.py]: (that belongs to a different PR, will not fix in this PR)

**Comment by berland** [comment]
This PR does not have any speed increases, so cannot comment on that.. It just enables concurrency by providing a general wrapper that results the Realization object after processing.

**Pull request merged**